extensions/everhour/src/views/RecentTaskList.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from "react";
+import { List, Icon, showToast, ToastStyle } from "@raycast/api";
+import { TaskListItem } from "../components";
+import { getRecentTasks, getCurrentTimer } from "../api";
+import { Task } from "../types";
+import { createResolvedToast } from "../utils";
+
+export function RecentTaskList() {
+  const [activeTimerTaskId, setActiveTimerTaskId] = useState<null | string>(null);
+  const [tasks, setTasks] = useState<Array<Task>>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const refreshActiveTimer = async () => {
+    const toast = await showToast(ToastStyle.Animated, "Refreshing tasks");
+    try {
+      const activeTimer = await getCurrentTimer();
+      setActiveTimerTaskId(activeTimer);
+      createResolvedToast(toast, "Tasks refreshed").success();
+    } catch (error) {
+      createResolvedToast(toast, "Failed to refresh tasks").error();
+    }
+  };
+
+  const fetchTasks = async () => {
+    const tasksResp = await getRecentTasks();
+    setTasks(tasksResp);
+  };
+
+  useEffect(() => {
+    async function fetch() {
+      const toast = await showToast(ToastStyle.Animated, "Fetching tasks");
+      try {
+        await fetchTasks();
+        setIsLoading(false);
+        createResolvedToast(toast, "Tasks fetched").success();
+      } catch (error) {
+        const message = (error as { message: string }).message;
+        createResolvedToast(toast, message || "Failed to fetch projects").error();
+        setIsLoading(false);
+      }
+    }
+    fetch();
```

Comment:
Having this as an IIFE might be better. Don't think it would be an issue but naming this function `fetch` could clash with the `global` fetch function.

```suggestion
(async function() {
  const toast = await showToast(ToastStyle.Animated, "Fetching tasks");
  try {
    await fetchTasks();
    setIsLoading(false);
    createResolvedToast(toast, "Tasks fetched").success();
  } catch (error) {
    const message = (error as { message: string }).message;
    createResolvedToast(toast, message || "Failed to fetch projects").error();
    setIsLoading(false);
  }
})();
```

Reply:
Will leave as is for now as it will otherwise produce conflicts with other changes I am working on -- approach to do this was taken from TaskList.tsx, so I'll refactor them together in follow ups if thats okay for you.